File: Java/Recursion-1/parenBit.java

public class recursion1 {
	
	// given a string that contains a single pair of parenthesis, compute recursively a new string made of only of the parenthesis and their contents, so "xyz(abc)123" yields "(abc)"
	public String parenBit(String str) {
		
		if (str.charAt(0) == ')') {
			return "";
		}
		if (str.charAt(0) == '(') {
			int index = str.indexOf(')');
			return str.substring(0, index+1) + parenBit(str.substring(index));
		}
		else {
			return parenBit(str.substring(1));
		}
		
	}

	public static void main(String[] args) {

		recursion1 o = new recursion1();
		
		System.out.println(o.parenBit("xyz(abc)123"));
		System.out.println(o.parenBit("x(hello)"));
		System.out.println(o.parenBit("(xy)1"));
		
	}

}

File: Python/Recursion-1/paren_bit.py

# given a string that contains a single pair of parenthesis, compute recursively a new string made of only of the parenthesis and their contents, so "xyz(abc)123" yields "(abc)"
def paren_bit(str):
	if str[0] == ')':
		return ''

	if str[0] == '(':
		index = str.index(')')
		return str[0:index+1] + paren_bit(str[index:])
	else:
		return paren_bit(str[1:])

print(paren_bit('xyz(abc)123'))
print(paren_bit('x(hello)'))
print(paren_bit('(xy)1'))
